src/braket/default_simulator/openqasm/interpreter.py

```diff
@@ -470,11 +470,13 @@ def _(self, node: QuantumGateModifier) -> QuantumGateModifier:
 
     @visit.register
     def _(self, node: QuantumMeasurement) -> None:
-        """Doesn't do anything, but may add more functionality in the future"""
+        qubits = self.context.get_qubits(node.qubit)
```

Comment:
you should visit `node.qubits` before calling get_qubits. For example if you have `q[i]` where is an int in scope, you will need to resolve that first. Try using `self.visit(node.qubits)` and that should resolve the issue you were having with range. I would also add a test for variables in a qubit index. Maybe
```
qubit[5] q;
for int i in [0:2] {
    measure q[i];
}
```
and assert measured qubits is [0, 1, 2]

Reply:
Added `self.visit(node.qubits)` before getting the qubits and added a test.